crates/router/src/utils.rs

```diff
@@ -1007,6 +1007,10 @@ pub async fn flatten_join_error<T>(handle: Handle<T>) -> RouterResult<T> {
     }
 }
 
+pub(crate) fn get_merchant_fingerprint_secret_key(merchant_id: &str) -> String {
+    format!("fingerprint_secret_{merchant_id}")
+}
```

Comment:
Nit: Can use `impl AsRef<str>` to allow passing `&str`, `&String` and `String`. Optional change, by the way.

```suggestion
pub(crate) fn get_merchant_fingerprint_secret_key(merchant_id: impl AsRef<str>) -> String {
    format!("fingerprint_secret_{}", merchant_id.as_ref())
}
```